Lib/test/test_timeit.py

```diff
@@ -387,6 +393,26 @@ def callback(a, b):
                     '500 0.488\n')
         self.assertEqual(s.getvalue(), expected)
 
+    def test_autorange_with_callback_and_total_time(self):
+        def callback(a, b):
+            print("{} {:.3f}".format(a, b))
+        with captured_stdout() as s:
+            num_loops, time_taken = self.autorange(callback=callback,
+                    total_time=0.6)
+        self.assertEqual(num_loops, 1000)
+        self.assertEqual(time_taken, 1000/1024)
+        expected = ('1 0.001\n'
```

Comment:
This test imports `dedent` so that might help?

```python
expected = dedent('''\
             1 0.001
             2 0.002
             5 0.005
             10 0.010
             20 0.020
             50 0.049
             100 0.098
             200 0.195
             500 0.488
             1000 0.977
''')
```

Reply:
Thanks for the hint.